samples/bluetooth/central_uart/src/main.c

```diff
@@ -289,14 +289,29 @@ static int uart_init(void)
 static void discovery_complete(struct bt_gatt_dm *dm,
 			       void *context)
 {
+	int err;
 	struct bt_nus_client *nus = context;
 	LOG_INF("Service discovery completed");
 
 	bt_gatt_dm_data_print(dm);
 
-	bt_nus_handles_assign(dm, nus);
-	bt_nus_subscribe_receive(nus);
+	err = bt_nus_handles_assign(dm, nus);
+	if (err)
+	{
+		bt_gatt_dm_data_release(dm);
```

Comment:
The `goto cleanup` and this line form a double-free, right?

Reply:
sorry, forgot to add these extra two releases once I changed it to the cleanup way.
fixed now